lib/launcher/src/Cardano/Launcher/Mithril.hs

```diff
@@ -0,0 +1,96 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Cardano.Launcher.Mithril
+    ( downloadLatestSnapshot
+    , downloadMithril
+    , MithrilExePath (..)
+    )
+    where
+
+import Prelude
+
+import qualified Data.ByteString as BS
+
+import Control.Monad.Extra
+    ( unlessM
+    )
+import Network.HTTP.Simple
+    ( getResponseBody
+    , httpBS
+    , parseRequest
+    )
+import System.Directory
+    ( doesFileExist
+    , withCurrentDirectory
+    )
+import System.FilePath
+    ( (</>)
+    )
+import System.Info
+    ( arch
+    , os
+    )
+import System.Process
+    ( callProcess
+    )
+
+newtype MithrilExePath = MithrilExePath { mithrilExePath :: FilePath }
+
+-- | Download the latest snapshot node db into /db relative to the supplied dir.
+downloadLatestSnapshot :: FilePath -> MithrilExePath -> IO ()
+downloadLatestSnapshot outputParentDir (MithrilExePath mithril) = do
+    callProcess mithril ["cdb", "download", "latest", "--download-dir", outputParentDir]
+
+-- | Downloads the latest Mithril release package,
+-- extracts it, and returns the path to the mithril client executable.
+--
+-- May interactively ask how to handle conflicts if items in the supplied
+-- working directory conflict with items in the mithril release archive.
```

Comment:
Maybe use `-U` option
```
     -U, --unlink, --unlink-first
             (x mode only) Unlink files before creating them.  This can be a minor performance optimization if most files already exist, but can make things
             slower if most files do not already exist.  This flag also causes tar to remove intervening directory symlinks instead of reporting an error.
             See the SECURITY section below for more details.
```

Reply:
I concidered using `--overwrite` but after making the working directory an empty temporary one I thought it was unnecessary, particularly if it could have a very slim chance of doing something unintended to your machine if we change this code around.